test-tool/test_prin_read_keys_simple.c

```diff
@@ -52,10 +52,10 @@ test_prin_read_keys_simple(void)
 
         logging(LOG_VERBOSE, "Test ADDITIONAL_LENGTH matches DATA_IN size.");
         al = scsi_get_uint32(&task->datain.data[4]);
-        if (al != task->datain.size - 8) {
```

Comment:
I think I messed up. When I was testing originally it always failed because it was expecting a non-zero value. I'm guessing my code returned non-zero in data[4] but I thought task->datain.size - 8 was incorrectly evaluating to non-zero.  I see zero for both now.